### ISSUE TITLE
Which apps use ARM TLS1.3?

### ISSUE THREAD

**Issue by berkayyildi** [tsarpaul/FBUnpinner#6]
Can you please explain which android devices or versions use ARM TLS1.3?
In addition to this I tested it with Facebook Version: 212.0.0.28.110 x86 TLS1.3 it does not work.
Also patcher do not work for old versions of facebook so you might be write which versions supported.

Thank you very much I am really excited to see ARM TLS1.3 and new usage documents.

**Comment by tsarpaul** [comment]
> Can you please explain which android devices or versions use ARM TLS1.3?

Hi, not sure about this question. ARM is a CPU architecture - most real life phones have an ARM CPU.
There are however Android devices that work on x86 - no idea which real life phones have this but there are emulators like Genymotion which are x86-based.

> In addition to this I tested it with Facebook Version: 212.0.0.28.110 x86 TLS1.3 it does not work.
The script should work on x86 212, maybe your sniffer setup is wrong?

> Also patcher do not work for old versions of facebook so you might be write which versions supported.
Which version?

**Comment by tsarpaul** [comment]
Added ARM32 support!

**Comment by berkayyildi** [comment]
- Hi again,I tested with old version of facebook with already patched library at another git repo so I am sure my sniffer is okey.
I tested _GenyMotion x86 Galaxy s9 API 26(8.0)_ with this apk:

> com.facebook.katana_212.0.0.28.110-145459263_minAPI26(x86)(280,360,400,420,480,560,640dpi)_apkmirror.com

Patching is done with TLS1.3 but when I replace, **It was not work again**.


- Thank you for ARM32 support but I tried with galaxy s4 I got error

> ERROR: **Failed to patch ARM TLS 1.3 stack, something has gone wrong!**

I think arm x32 patch is not stable. I hope you to fix it.


- Also Then I tried with old x86 facebook apk:

> facebook-189-0-0-35-91-3 (September 11, 2018)

I got error **Could not find the required code to patch!**
